pkgs/tools/inputmethods/keymapper/default.nix

```diff
@@ -2,34 +2,47 @@
 , stdenv
 , fetchFromGitHub
 , cmake
+, asio
 , dbus
 , libX11
 , libusb1
 , pkg-config
 , udev
 , wayland
+, darwin
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "keymapper";
-  version = "3.0.0";
+  version = "3.5.3";
 
   src = fetchFromGitHub {
     owner = "houmain";
     repo = "keymapper";
     rev = finalAttrs.version;
-    hash = "sha256-X2Qk/cAczdkteB+6kyURGjvm1Ryio6WHj3Ga2POosCA=";
+    hash = "sha256-CfZdLeWgeNwy9tEJ3UDRplV0sRcKE4J6d3CxC9gqdmE=";
   };
 
-  # all the following must be in nativeBuildInputs
+  patches = lib.optional stdenv.isDarwin [
```

Comment:
```suggestion
  patches = lib.optionals stdenv.isDarwin [
```

Reply:
There's a difference between `lib.optional` and `lib.optionals`.

In this case, I will require to change to `lib.optionals`.

Reply:
Thanks. Didn't notice that.